src/prism.c

```diff
@@ -19501,7 +19501,13 @@ parse_expression_prefix(pm_parser_t *parser, pm_binding_power_t binding_power, b
                 pm_do_loop_stack_push(parser, false);
                 statements = (pm_node_t *) pm_statements_node_create(parser);
 
-                pm_node_t *statement = parse_expression(parser, PM_BINDING_POWER_DEFINED + 1, binding_power < PM_BINDING_POWER_COMPOSITION, false, PM_ERR_DEF_ENDLESS, (uint16_t) (depth + 1));
+                // In endless method bodies, we need to handle command calls carefully.
```

Comment:
I do want to mention that the failed test mentions it being a technical limitation of parse.y: https://github.com/ruby/ruby/blob/86320a53002a3adaf35ad7434c70e86747a8b345/test/ruby/test_syntax.rb#L1797 (also https://bugs.ruby-lang.org/issues/17398#note-5). Perhaps add a comment that this is to accomodate the overly specific ruby/ruby test? It looks like this was supposed to be valid syntax but it was too difficult to implement.

Reply:
+1 @st0012 do you mind adding a comment?  🙇

Reply:
Updated 👍